### PR TITLE
feat(components): you can now place a close button within the popover

### LINKED ISSUE #2270: [BUG] Popover not working open controll When using backdrop option on touch devie
### NextUI Version

2.2.9

### Describe the bug

I found a bug regarding popover not working open controll When using backdrop option on touch devie.
I wanted to place a close button inside the popover and tried to implement it using useState.
It worked well on a PC, but did not work properly on a touch device.
This only happens when the Backdrop option is set.

### Your Example Website or App

No response

### Steps to Reproduce the Bug or Issue

A sample code is shown below.
```jsx
import React from 'react'
import {
  Popover,
  PopoverTrigger,
  PopoverContent,
  Button,
} from '@nextui-org/react'

export default function App() {
  const [isOpen, setIsOpen] = React.useState(false)

  return (
    <div className='flex flex-col gap-2'>
      <Popover
        isOpen={isOpen}
        onOpenChange={(open) => {
          console.log('onOpenChange argument is', open)
          setIsOpen(open)
        }}
        backdrop='opaque'
      >
        <PopoverTrigger>
          <Button>OpenState: {isOpen ? 'Open' : 'Close'}</Button>
        </PopoverTrigger>
        <PopoverContent>
          <div className='px-1 py-2'>
            <div className='text-small font-bold'>Popover Content</div>
            <Button
              color='danger'
              variant='light'
              aria-label='Close'
              onPress={() => {
                setIsOpen(false)
              }}
            >
              Close
            </Button>
          </div>
        </PopoverContent>
      </Popover>
    </div>
  )
}
```

### Expected behavior

When processing the close button with the content in the popover
the next time the popover is opened, it is closed immediately.
When I check in the console, I see that onOpenChange is called twice.

### Screenshots or Videos

_No response_

### Operating System Version

macOS

### Browser

Chrome

### PR DESCRIPTION
**Pull request by kght6123**
スクリーンショット 2024-01-22 23 08 37

Closes #2270

## 📝 Description

You can now place a close button within the popover.

## ⛳️ Current behavior (updates)

Currently, when a function is specified in `children`, it only has `titleProps` as an argument. Therefore, I have added `onClose` as a second argument.

## 🚀 New behavior

By adding `onClose` to the arguments of the function that can be passed to `children` in `popover-content.tsx`, you can now add a close process to any button or similar element.

I have also corrected the onClose of the DismissButton, which is added for accessibility, since its closing process is different from that of clicking the background of the Popover. This difference might cause issues on touch devices as mentioned in the issue.
https://react-spectrum.adobe.com/react-aria/usePopover.html#example

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

Nothing.

## Summary by CodeRabbit

- **New Features**
	- Introduced the ability to add a close button within the popover's content, enhancing user control.
	- Added a new story template featuring a popover with an `onClose` button for better demonstration and testing.

- **Enhancements**
	- Improved interaction by enabling the closing of popovers when the backdrop is clicked, through a new dedicated callback function.

- **Refactor**
	- Enhanced the `PopoverContent` component to support passing an `onClose` function, improving component flexibility and functionality.

### CODE REVIEW

File: .changeset/quick-needles-cross.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/popover": major
```
Comment (jrgarciadev) [inline .changeset/quick-needles-cross.md:2]: @kght6123 

```suggestion
"@nextui-org/popover": patch
```

Update the changeset to be a `patch`, when creating it just press enter without selecting that it is a `major` / `minor`

**Review by jrgarciadev — CHANGES_REQUESTED** [review]

**Comment by wingkwong** [comment]
@kght6123 can you resolve the conflicts and add some tests?

**Comment by kght6123** [comment]
@wingkwong 
The dispute has been resolved. Please check.

What are tests? Where and what kind of tests should I add?
Due to the current busy schedule, the fix may take until the end of May.
We apologize for the inconvenience, but please let us know.

**Comment by wingkwong** [comment]
@kght6123 See `packages/components/popover/__tests__/popover.test.tsx`. You should add the test to cover the logic in introduced in this PR.

**Comment by kght6123** [comment]
@wingkwong I've got it. thank you.

**Comment by wingkwong** [comment]
Closing due to inactivity.

**Pull request closed without merging**